src/TpetraLinearSystem.C

```diff
@@ -155,28 +155,28 @@ int TpetraLinearSystem::getDofStatus(stk::mesh::Entity node)
   bool has_non_matching_boundary_face_alg = false;
   bool hasPeriodic = false;
 
-  bool debug = false;
+  if (NULL == realm_.allPeriodicInteractingSelector_ && realm_.allPeriodicInteractingParts_.size())
```

Comment:
I used a Selector that is created once for efficiency reasons - I just assumed that selectUnion was expensive, and this method gets called for every node.  It would be cleaner to only store the PartVector, however.